mixer/pkg/expr/finder.go

```diff
@@ -0,0 +1,70 @@
+// Copyright 2018 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package expr
+
+import (
+	"fmt"
+	"sort"
+
+	configpb "istio.io/istio/mixer/pkg/config/proto"
+	"istio.io/istio/mixer/pkg/pool"
+)
+
+// EmptyFinder is a AttributeDescriptorFinder with no attributes
+var EmptyFinder AttributeDescriptorFinder = finder{
```

Comment:
where is empty finder used?
even without allocating an zero length map, you should be able to get the same `GetAttribute()` behaviour.

Reply:
In config declaration mainly. https://github.com/ozevren/istio/blob/runtime2-config/mixer/pkg/runtime2/config/snapshot.go This is the only place it is used. I'll apply the same implicit map pattern there as well and remove the explicit initializations.

Thanks!